### ISSUE TITLE
Missing labels and annotations after last step of canary rollout

### ISSUE THREAD

**Issue by Yorkerrr** [argoproj/argo-rollouts#883]
## Summary 

I want to implement a canary release with istio as a traffic manager. Everything works as expected but in a result replica-set istio-related annotations and labels are missing. Here is a diff between the initial pod manifest and a final one:

<img width="960" alt="Screen Shot 2020-12-07 at 5 42 20 PM" src="https://user-images.githubusercontent.com/1927829/101427359-90870000-38b3-11eb-84d3-6b8eed8a81a6.png">

## Diagnostics

argo-rollouts:v0.10.1
argocd:v1.7.9
istio:v1.7.3

```
spec:
  strategy:
    canary:
      canaryMetadata:
        labels:
          role: canary
      stableMetadata:
        labels:
          role: stable
      analysis:
        templates:
         - templateName: success-rate
        startingStep: 2
        args:
        - name: service-name
          value: demo-canary.canary.svc.cluster.local
      steps:
      - setWeight: 20
      - pause:
          duration: "15s"
      - setWeight: 50
      - pause:
          duration: "1m"
      canaryService: demo-canary
      stableService: demo
      trafficRouting:
        istio:
           virtualService: 
            name: istio-vs
            routes:
            - primary 
```

# Paste the logs from the rollout controller
```
2020-12-08T01:35:46.541924016Z time="2020-12-08T01:35:46Z" level=info msg="Started syncing rollout at (2020-12-08 01:35:46.54067444 +0000 UTC m=+9900.077589128)" namespace=canary rollout=demo
2020-12-08T01:35:46.543161892Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=canary rollout=demo
2020-12-08T01:35:46.543174592Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling StableRS" namespace=canary rollout=demo
2020-12-08T01:35:46.543180793Z time="2020-12-08T01:35:46Z" level=info msg="No StableRS exists to reconcile or matches newRS" namespace=canary rollout=demo
2020-12-08T01:35:46.543186893Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling new ReplicaSet 'demo-799cb5ffc6'" namespace=canary rollout=demo
2020-12-08T01:35:46.543192593Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling old replica sets" namespace=canary rollout=demo
2020-12-08T01:35:46.543198294Z time="2020-12-08T01:35:46Z" level=info msg="No Steps remain in the canary steps" namespace=canary rollout=demo
2020-12-08T01:35:46.543513213Z time="2020-12-08T01:35:46Z" level=info msg="Rollout has executed every step" namespace=canary rollout=demo
2020-12-08T01:35:46.543523414Z time="2020-12-08T01:35:46Z" level=info msg="New RS has successfully progressed" namespace=canary rollout=demo
2020-12-08T01:35:46.543750527Z time="2020-12-08T01:35:46Z" level=info msg="No status changes. Skipping patch" namespace=canary rollout=demo
2020-12-08T01:35:46.543762428Z time="2020-12-08T01:35:46Z" level=info msg="Reconciliation completed" namespace=canary rollout=demo time_ms=2.932179
2020-12-08T01:35:46.549559681Z time="2020-12-08T01:35:46Z" level=info msg="Started syncing Analysis at (2020-12-08 01:35:46.549436874 +0000 UTC m=+9900.086351562)" analysisrun=demo-799cb5ffc6-2 namespace=canary
2020-12-08T01:35:46.549847599Z time="2020-12-08T01:35:46Z" level=info msg="No status changes. Skipping patch" analysisrun=demo-799cb5ffc6-2 namespace=canary
2020-12-08T01:35:46.549856799Z time="2020-12-08T01:35:46Z" level=info msg="Reconciliation completed" analysisrun=demo-799cb5ffc6-2 namespace=canary time_ms=0.33642
2020-12-08T01:35:46.554817301Z time="2020-12-08T01:35:46Z" level=info msg="Started syncing rollout at (2020-12-08 01:35:46.554163561 +0000 UTC m=+9900.091078249)" namespace=canary rollout=demo
2020-12-08T01:35:46.555916368Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=canary rollout=demo
2020-12-08T01:35:46.555941570Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling StableRS" namespace=canary rollout=demo
2020-12-08T01:35:46.555946070Z time="2020-12-08T01:35:46Z" level=info msg="No StableRS exists to reconcile or matches newRS" namespace=canary rollout=demo
2020-12-08T01:35:46.556012374Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling new ReplicaSet 'demo-799cb5ffc6'" namespace=canary rollout=demo
2020-12-08T01:35:46.556024975Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling old replica sets" namespace=canary rollout=demo
2020-12-08T01:35:46.556029075Z time="2020-12-08T01:35:46Z" level=info msg="No Steps remain in the canary steps" namespace=canary rollout=demo
2020-12-08T01:35:46.556352695Z time="2020-12-08T01:35:46Z" level=info msg="Rollout has executed every step" namespace=canary rollout=demo
2020-12-08T01:35:46.556368296Z time="2020-12-08T01:35:46Z" level=info msg="New RS has successfully progressed" namespace=canary rollout=demo
2020-12-08T01:35:46.556504204Z time="2020-12-08T01:35:46Z" level=info msg="No status changes. Skipping patch" namespace=canary rollout=demo
2020-12-08T01:35:46.556510204Z time="2020-12-08T01:35:46Z" level=info msg="Reconciliation completed" namespace=canary rollout=demo time_ms=2.277439
2020-12-08T01:35:46.566771129Z time="2020-12-08T01:35:46Z" level=info msg="Started syncing rollout at (2020-12-08 01:35:46.565116828 +0000 UTC m=+9900.102031416)" namespace=canary rollout=demo
2020-12-08T01:35:46.571145695Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling TrafficRouting with type 'Istio'" namespace=canary rollout=demo
2020-12-08T01:35:46.571305305Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling StableRS" namespace=canary rollout=demo
2020-12-08T01:35:46.571396911Z time="2020-12-08T01:35:46Z" level=info msg="No StableRS exists to reconcile or matches newRS" namespace=canary rollout=demo
2020-12-08T01:35:46.571558921Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling new ReplicaSet 'demo-799cb5ffc6'" namespace=canary rollout=demo
2020-12-08T01:35:46.571651426Z time="2020-12-08T01:35:46Z" level=info msg="Reconciling old replica sets" namespace=canary rollout=demo
2020-12-08T01:35:46.571765733Z time="2020-12-08T01:35:46Z" level=info msg="No Steps remain in the canary steps" namespace=canary rollout=demo
2020-12-08T01:35:46.572361569Z time="2020-12-08T01:35:46Z" level=info msg="Rollout has executed every step" namespace=canary rollout=demo
2020-12-08T01:35:46.572464376Z time="2020-12-08T01:35:46Z" level=info msg="New RS has successfully progressed" namespace=canary rollout=demo
2020-12-08T01:35:46.572740492Z time="2020-12-08T01:35:46Z" level=info msg="No status changes. Skipping patch" namespace=canary rollout=demo
2020-12-08T01:35:46.572843599Z time="2020-12-08T01:35:46Z" level=info msg="Reconciliation completed" namespace=canary rollout=demo time_ms=7.666367

# Logs for the entire controller:
kubectl logs -n argo-rollouts deployment/argo-rollouts

# Logs for a specific rollout:
kubectl logs -n argo-rollouts deployment/argo-rollouts | grep rollout=<ROLLOUTNAME>
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by jessesuen** [comment]
This is a bug. The `canaryMetadata` and `stableMetadata` feature is dropping the injected labels/annotations on the pods. Will prioritize a fix for a v0.10.2 patch.

**Comment by jessesuen** [comment]
Fixed in v0.10.2 (now released)
